### ISSUE TITLE
Create context section in components

### ISSUE THREAD

**Issue by fawadasaurus** [candlecorp/wick#212]
**Is your feature request related to a problem? Please describe.**
Sometimes the input for operations will just be static values.  These will only be defined in design time (ex: a regex pattern). These values will then need to be referenced cleanly in the flow.

~~**Propose a solution**~~
~~There should be a section in the composite component to define `constants`.  these can be a key-value pair that creates named values that can be referenced in the flow.~~

```yaml
...
constants:
  - name: regex-pattern
    type: string
    value: "(.+)"
  - name: header-row
     type: bool
     value: false
 ...~~
 
flow:
  - <> -> regex.input
  - const.regex-pattern -> regex.pattern
``` 

~~**Describe alternatives you've considered**~~
~~I have considered the idea of putting the string directly in the flow but that would make the flow difficult to read and difficult to support multi-line strings.~~

~~**Additional context**~~
~~None~~

**Comment by fawadasaurus** [comment]
After further discussion, it is clear that constants are not needed as these types of uses are `configs` for components that will likely not be runtime generated but known in design time.  For this, we should change to add a `config` element for components so they can have initialization values that are created at design time and immutable during runtime.


```yaml
---
name: "regex"
kind: wick/component@v1
metadata:
  description: "A regular expression component for matching and capturing strings."
  authors: ["Candle Corporation"]
  licenses: ["Apache-2.0"]
  documentation: Source code available at https://github.com/candlecorp/wick-components/tree/main/components
  version: "0.0.2"
component:
  kind: wick/component/wasmrs@v1
  ref: build/regex.signed.wasm
  operations:
    - name: rematch
      configs:
        - name: pattern
           type: string
      inputs:
        - name: input
          type: string
        - name: pattern
          type: string
      outputs:
        - name: result
          type: bool
        - name: matches
          type: { list: { type: string } }
          optional: true
tests:
  - name: regex
    operation: rematch
    with:
      - name: pattern
         data: '(\w+)' # match words
    input:
      - name: input
        data: "This is a string.  It is not very long."
    output:
      - name: result
        data: true
      - name: matches
        data: '[["This"], ["is"], ["a"], ["string"], ["It"], ["is"], ["not"], ["very"], ["long"]]'

```

In the lib.rs, here is a way to reference the config:

```rust
async fn rematch(
        mut input: WickStream<String>,
        mut outputs: OpRematchOutputs,
        ctx: OpRematchCtx, //All components should be passed context.  The context is values that are sent upon each operation invocation. 
    ) -> wick::Result<()> {
   let pattern = ctx.configs.pattern.as_str();
    //code here
    
}
```

**Comment by jsoverson** [comment]
Fixed via #226
